web/federate.go

```diff
@@ -161,15 +165,21 @@ func (h *Handler) federation(w http.ResponseWriter, req *http.Request) {
 	sort.Strings(externalLabelNames)
 
 	var (
-		lastMetricName string
-		protMetricFam  *dto.MetricFamily
+		lastMetricName   string
+		lastWasHistogram bool
+		protMetricFam    *dto.MetricFamily
 	)
 	for _, s := range vec {
+		isHistogram := s.H != nil
+		if s.H != nil &&
+			format != expfmt.FmtProtoDelim && format != expfmt.FmtProtoText && format != expfmt.FmtProtoCompact {
+			// Can't serve the native histogram.
```

Comment:
Perhaps add a TODO so we find it once the other protocols support native histograms?